subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/CancellableConsumerConnection.java

```diff
@@ -106,4 +120,12 @@ private CancellableActionRunner(InternalCancellableConnection executor, Transfor
             return result.getModel();
         }
     }
+    /**
+     * Interface representing either an {@link InternalCancellableConnection}
+     * or an {@link org.gradle.tooling.internal.protocol.InternalCancellableConnectionVersion2}.
+     */
+    protected interface InternalCancellableConnectionWrapper {
```

Comment:
Should not be necessary. Instead, the new connection should use a subclass of CancellableActionRunner which then also supports the new parameterized actions.